Lib/http/client.py

```diff
@@ -211,7 +211,7 @@ def parse_headers(fp, _class=HTTPMessage):
         if line in (b'\r\n', b'\n', b''):
             break
     hstring = b''.join(headers).decode('iso-8859-1')
-    return email.parser.Parser(_class=_class).parsestr(hstring)
+    return email.parser.Parser(_class=_class).parsestr(hstring, headersonly=True)
```

Comment:
For a change like this, please

1) Please create  bugs.python.org ticket, and show us the problem with steps to reproduce.
2) Show how this fixes along with a tests.
3) We need to document this too. 

This patch is incomplete (even for review) without all three above.